kernel-rs/src/virtio/virtio_disk.rs

```diff
@@ -352,6 +375,210 @@ impl VirtioDisk {
         guard.wakeup(ctx.kernel());
     }
 
+    // This method writes the data in the given buffers to disk sequentially, therefore,
+    // minimizing seek-time overhead of the disk. It is designed for using with `Lfs`.
+    // Calling this method when using `Ufs` is effectively not different from calling
+    // `VirtioDisk::rw` and may incur additional performance overhead.
+    fn write_seq(
+        guard: &mut SleepableLockGuard<'_, Self>,
+        barray: &mut [Option<Buf>; MAX_SEQ_WRITE],
+        ctx: &KernelCtx<'_, '_>,
+    ) {
+        // Maximum # of blocks to be sequentially written to the disk must be less than
+        // the total # of available descriptors excluding two descriptors, which will be
+        // allocated for the header and tailer descriptors.
+        const_assert!(MAX_SEQ_WRITE + 2 <= NUM);
+
+        let mut darray: [Option<Descriptor>; MAX_SEQ_WRITE] = array![_ => None; MAX_SEQ_WRITE];
```

Comment:
I'm curious if this can be [Descriptor; MAX_SEQ_WRITE] not [Option<Descriptor>; MAX_SEQ_WRITE]